vignettes/creating-custom-modules.Rmd

```diff
@@ -15,170 +15,230 @@ vignette: >
 
-### Server function
+The server function is where the main logic of a `teal` module is handled. 
+For our histogram module, the server function will handle user interactions and manage the reactive `teal_data` object, which allows the module to dynamically respond to user inputs.
 
-This function contains the `shiny` server logic for the module and should be of the form:
+### Passing the `data` Argument to the Server Function
 
-```{r, eval=FALSE}
-function(
-    id,
-    data, # optional; use if module needs access to application data
-    filter_panel_api, # optional; use if module needs access to filter panel; see teal.slice
-    reporter, # optional; use if module supports reporting; see reporting vignette
-    ...) {
+To begin, it’s essential to include the `data` argument in the server function definition. 
+
+This `data` argument holds the reactive `teal_data` object, which contains your datasets and any filters applied. Including `data` ensures:
+
+- The server function receives a reactive version of `teal_data`, allowing it to automatically respond to changes.
+- The server can access the filtered datasets directly.
+
+The correct function definition for the server function is:
+
+```r
+histogram_module_server <- function(id, data) {
   moduleServer(id, function(input, output, session) {
-    # module code here
+    # Server logic goes here
   })
 }
 ```
 
-The data that arrives in the module is a `teal_data` object, the data container used throughout the `teal` application.
-`teal_data` is passed to the `init` function when building the application and, after filtering by the filter panel, it is passed to modules, wrapped in a reactive expression.
-The `teal_data` class allows modules to track the `R` code that they execute so that module outputs can be reproduced.
-See the `teal.data` package for a [detailed explanation](https://insightsengineering.github.io/teal.data/latest-tag/articles/teal-data-reproducibility.html).
+### Understanding `teal_data` as a Reactive Object in Server Logic
 
-## Example modules
+When used in the server logic of a `teal` module, the `teal_data` object becomes a **reactive data container**.
+This means that to access its contents, you need to call it like a function, using parentheses: `data()`.
 
-### Viewing data
+This syntax triggers reactivity, ensuring that the data within `teal_data` stays up-to-date with any filters or changes applied elsewhere in the application.
 
-Here is a minimal module that allows the user to select and view one dataset at a time.
-By default, filtering is enabled for all datasets.
-Note that dataset choices are specified by the `datanames` property of the `teal_data` container.
+> **Note**: The `teal_data` object behaves as a reactive data container only when used within the server logic. If accessed outside of the server, it will not be reactive.
 
-```{r, message=FALSE}
-library(teal)
+### Using `datanames()` to Access Dataset Names in `teal_data` object
```

Comment:
I will need to change this to just using `names()` once `datanames()` is deprecated.

Reply:
Done with 5617508983db5686177b3260a469fdbb71a0e6fb